tensorboard/uploader/uploader.py

```diff
@@ -224,52 +236,77 @@ def send_requests(self, run_to_events):
           RuntimeError: If no progress can be made because even a single
           point is too large (say, due to a gigabyte-long tag name).
         """
+        # TODO(soergel): Allow enabling/disabling upload per plugin
 
-        for (run_name, event, orig_value) in self._run_values(run_to_events):
-            value = data_compat.migrate_value(orig_value)
-            time_series_key = (run_name, value.tag)
-
-            # The metadata for a time series is memorized on the first event.
-            # If later events arrive with a mismatching plugin_name, they are
-            # ignored with a warning.
-            metadata = self._tag_metadata.get(time_series_key)
-            if metadata is None:
-                metadata = value.metadata
-                self._tag_metadata[time_series_key] = metadata
-
-            if value.HasField("metadata") and (
-                value.metadata.plugin_data.plugin_name
-                != metadata.plugin_data.plugin_name
-            ):
-                logger.warning(
-                    "Mismatching plugin names for %s.  Expected %s, found %s.",
-                    time_series_key,
-                    metadata.plugin_data.plugin_name,
-                    value.metadata.plugin_data.plugin_name,
-                )
-            elif (
-                metadata.plugin_data.plugin_name == scalar_metadata.PLUGIN_NAME
-            ):
-                self._scalar_request_sender.add_event(
-                    run_name, event, value, metadata
-                )
-            # TODO(nielsene): add Tensor plugin cases here
-            # TODO(soergel): add Graphs blob case here
+        for (run_name, event) in self._run_events(run_to_events):
+            if event.graph_def or event.meta_graph_def:
+                self._send_graph(run_name, event)
```

Comment:
This should be using `dataclass_compat.migrate_event` rather than poking
into the event directly. The whole point of `dataclass_compat` is to
enable the upload flow to work in the same way as the local multiplexer
flow. In particular, that will set the correct tag name here, which is
required for the frontend to request the correct graph.

Reply:
Ah, right, thanks for unconfusing me.  Doing that required rejiggering things to be entirely generic-data-flavored, which was a) necessary anyway and b) surprisingly straightforward :)